msu/systems/mod_settings/settings_element.nut

```diff
@@ -0,0 +1,109 @@
+::MSU.Class.SettingsElement <- class
+{
+	Name = null;
+	ID = null;
+	static Type = "Element";
+	Flags = null;
+	Description = null;
+	Parent = null;
+
+	constructor(_id, _name = null)
+	{
+		if (_id.find(" ") != null)
+		{
+			::logError("The ID of a Settings Element should not have spaces");
+			throw ::MSU.Exception.InvalidType;
+		}
+		this.Name = _name == null ? _id : _name;
+		this.ID = _id;
+		this.Flags = [];
+		this.Description = "";
+	}
+
+	function getType()
+	{
+		return this.Type;
+	}
+
+	function setParent( _parent )
+	{
+		this.Parent = _parent.weakref();
+	}
+
+	function getParent()
+	{
+		return this.Parent;
+	}
+
+	function getPanelID()
+	{
+		return this.getParent().getParent().getID();
```

Comment:
Maybe the parent should have a `getPanelID` function too and this line should instead read `this.getParent().getPanelID()`;

Reply:
Agreed